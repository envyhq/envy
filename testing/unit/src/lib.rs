use unicode_segmentation::UnicodeSegmentation;

// TODO: we dont really need this package, just move this to lexer package's tests mod
pub fn str_to_graphemes(input: &str) -> Vec<String> {
    input
        .trim()
        .graphemes(true)
        .map(|char| char.to_owned())
        .collect::<Vec<_>>()
}

#[cfg(test)]
mod tests {
    use super::str_to_graphemes;

    #[test]
    fn converts_str_to_grapheme_vector() {
        let input = "
hello 😍 我 name is a̐ é ö̲
wow
cool
";

        let output = str_to_graphemes(input);

        assert_eq!(output.len(), 32);
        insta::assert_debug_snapshot!(output);
    }
}
