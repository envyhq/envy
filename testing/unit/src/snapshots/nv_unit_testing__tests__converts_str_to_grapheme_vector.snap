---
source: testing/unit/src/lib.rs
expression: output
---
[
    "h",
    "e",
    "l",
    "l",
    "o",
    " ",
    "😍",
    " ",
    "我",
    " ",
    "n",
    "a",
    "m",
    "e",
    " ",
    "i",
    "s",
    " ",
    "a\u{310}",
    " ",
    "e\u{301}",
    " ",
    "o\u{308}\u{332}",
    "\n",
    "w",
    "o",
    "w",
    "\n",
    "c",
    "o",
    "o",
    "l",
]
